src/coreclr/jit/objectalloc.cpp

```diff
@@ -347,17 +347,25 @@ void ObjectAllocator::ComputeStackObjectPointers(BitVecTraits* bitVecTraits)
                     {
                         // Check if we know what is assigned to this pointer.
                         unsigned bitCount = BitVecOps::Count(bitVecTraits, m_ConnGraphAdjacencyMatrix[lclNum]);
-                        assert(bitCount <= 1);
-                        if (bitCount == 1)
+                        if (bitCount > 0)
```

Comment:
Do you recall why you changed the logic here?

Reply:
IIRC it was done in 39d1ad9a2ba71031550f92259c1626e0ec497404, to resolve Pri0 test failures.

Reply:
Let's see how CI is going after bringing back the assertion.

Reply:
I just checked `b89546`, we end up the following gentree:
```
***** BB02 [0001]
STMT00002 ( 0x004[E-] ... 0x00B )
               [000009] DA-XG------                         *  STORE_LCL_VAR long   V01 loc1
               [000008] ---XG------                         \--*  INDEX_ADDR byref RangeTest[]
               [000006] -----------                            +--*  LCL_VAR   ref    V00 loc0
               [000007] -----------                            \--*  LCL_VAR   long   V02 loc2
```
which points both `V00` and `V02` to `V01`, and causes `bitCount` to be `2`, which seems incorrect.
We should only point `V00` to `V01` in this case instead. I'll push a fix to this.